core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -135,16 +142,41 @@ public long requestCount() {
     }
 
     this.opts = opts;
-    if (opts.indexCacheSize > 0) {
-      this.indexCache = new LruBlockCache(opts.indexCacheSize, CACHE_BLOCK_SIZE);
-    } else {
-      this.indexCache = new NoopCache();
-    }
 
-    if (opts.dataCacheSize > 0) {
-      this.dataCache = new LruBlockCache(opts.dataCacheSize, CACHE_BLOCK_SIZE);
-    } else {
-      this.dataCache = new NoopCache();
+    if (opts.indexCacheSize > 0 || opts.dataCacheSize > 0) {
+      ConfigurationCopy cc = new ConfigurationCopy();
```

Comment:
RFileScanner has `opts.tableConfig` which allows the user to set accumulo properties.  Could use this create this cache.  This would allow a user to change the cache factory impl.